### DISCUSSION TITLE
On the JIRA FlowViz - need for more fields

### DISCUSSION THREAD

**Discussion by Karloyski** [nbrown02/FlowViz-Jira#41] [Q&A] [answered]
Hi Nick,
I went over several discussions hoping to find the answer and found one closest which is data pulled into power BI. I did tried to expand the tables, but just could not get to to it. I would like to add more info into the tooltips for the trhough put visulaitio, I want to add information around sprint name and story points. Can you possibly give me guidance on this? Thank you!

**Comment by nbrown02** [comment, marked as answer]
Note: whilst these steps are applied for a sprint name, they can also be applied to fields such as team name or any other fields you wish to filter on!

1. The first thing you need to do is figure out what those fields are referred to in the API on your specific Jira instance. 
The field name in my Jira instance won't be the same as yours. To do this, find an item assigned to a particular sprint, in this instance I'm using CIT-1695 assigned to Sprint 70:

![image](https://github.com/user-attachments/assets/aa9117d9-0db6-491b-ad33-c2cc05d31182)

2. In the web browser, enter into the URL:

https:// ADD YOUR JIRA INSTANCE HERE.atlassian.net/rest/api/2/search?jql=issue%20%3D%20ADD THE ISSUE ID HERE

So for my issue that's:

https://jirainstanceforourorg.atlassian.net/rest/api/2/search?jql=issue%20%3D%20CIT-1695

3. Do a CTRL+F to find the sprint, (in this example 'sprint 70') and find the customfield name:
![image](https://github.com/user-attachments/assets/4e2f7dfe-0ae8-44fb-817b-3608762b0c41)

Write this down as we'll need it :)

4. In Power BI in your report, go into Power Query by clicking 'Transform Data'

![image](https://github.com/user-attachments/assets/ae504308-46d4-431f-9b30-486bb5b10886)

5. Go to _WorkItems Completed_ and change the 'Source' to add in your custom field

<img width="1002" alt="image" src="https://github.com/user-attachments/assets/718f4161-246c-4b74-b850-184810888074" />

6. Go to your _StartAt_ function and ensure this also has the custom field added

<img width="1135" alt="image" src="https://github.com/user-attachments/assets/c3e74dd6-7a47-4cd1-9da4-d5a694e0e149" />

7. Go to the 'Expanded Data.Fields' applied step

![image](https://github.com/user-attachments/assets/80cf0f17-cbe3-4182-8be8-185f0d0f50c2)

8. Add in the following to the formula bar:

![image](https://github.com/user-attachments/assets/f10ebafc-d674-4cfa-af49-3b2d1c8d06fc)

9. Do the following:

![Sprint](https://github.com/user-attachments/assets/823e9dae-7fbc-4dfd-a9c1-d2a92e7c92f0)

10. You'll then have the sprint details. You need to repeat this approach for _WorkItems InProgress_ (updating the _StartAt2_ function) table

11. For the _WorkItems History_, _WorkItems History5_, _WorkItems WIP_, _WorkItems WIP 2_ and _WorkItems Currently Blocked_ tables, you need to repeat steps 7-9.

12. For the formula for your calculated table of WorkItems WIP 3, you will need to update this to:
```
WorkItems WIP 3 = 
    UNION (
        SELECTCOLUMNS (
            'WorkItems WIP',
            "CompletedDate", [CompletedDate],
            "Date", [Date],
            "DateSK", [DateSK],
            "InProgressDate", [InProgressDate],
            "IssueID", [IssueID],
            "IssueType", [IssueType],
            "Today",[Today],
            "WIPAge",[WIPAge],
            "Sprint",[Sprint],
            "WIPAgeGrouped",[WIPAgeGrouped]
        ),
        SELECTCOLUMNS (
            'WorkItems WIP 2',
            "CompletedDate", [CompletedDate],
            "Date", [Date],
            "DateSK", [DateSK],
            "InProgressDate", [InProgressDate],
            "IssueID", [IssueID],
            "IssueType", [IssueType],
            "Today",[Today],
            "WIPAge",[WIPAge],
            "Sprint",[Sprint],
            "WIPAgeGrouped",[WIPAgeGrouped]
        )
    )
```

13. Assuming you want this as a slicer, you will need to create a calculated table like so:

```
zzSprint = 
DISTINCT (
    UNION (
        SELECTCOLUMNS (
            'WorkItems Completed',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems History',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems History2',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems History3',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems History4',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems History5',
            "Sprint", [Sprint]
        ),
        SELECTCOLUMNS (
            'WorkItems InProgress',
            "Sprint", [Sprint]
        )
    )
)
```

14. You will then need to go Modelling > Manage Relationships and create a One-to-many relationship with all your WorkItems tables and this new zzSprint table.

<img width="916" alt="image" src="https://github.com/user-attachments/assets/60eec015-701b-4aa1-9797-a171551343a7" />

15. Then you will be ready to use the new sprint table and its values as a slicer in your data

**Reply by Karloyski** [reply]
Wow! This helped a lot. I struggled finding that custom field at first, but with this detailed guide, I was able to get it. Thank you very much again!
